### PR TITLE
S of S.O.L.I.D principle satisfied

### PR DESCRIPTION
**Pull request by Muhammadali2025222**
Added comments and created a class that now handles inheritance from abstract classes and changes in child classes, main and shapefactory are made

### CODE REVIEW

**Comment by abbasshah17** [comment]
1. Change your comments to: "This class follows the Single Responsibility Principle.
2. Mention the responsibility of the class as you understand it.

**Comment by abbasshah17** [comment]
Still missing the description of the responsibility of each class in the comments.

Before deciding whether or not a class is following the Single Responsibility principle, first you need to understand the responsibility of the class or define it.

Once you've defined the responsibility of the class you can then assess whether or not the class follows the single responsibility principle or not.

**Review by abbasshah17 — CHANGES_REQUESTED** [review]
Still missing the description of the responsibility of each class in the comments.

Before deciding whether or not a class is following the Single Responsibility principle, first you need to understand the responsibility of the class or define it.

Once you've defined the responsibility of the class you can then assess whether or not the class follows the single responsibility principle or not.

**Comment by Muhammadali2025222** [comment]
I have suggestion that we create separate classes to override these methods.
For example
FOR CIRCLE: a circle drawer class a circle input class and then a circle volume and a circle area.
FOR RECTANGLE: a Rectangle drawer class a Rectangle input class and then a Rectangle volume and a Rectangle area.
FOR SQUARE: a Square drawer class a Square input class and then a Square volume and a Square area.
FOR TRIANGLE: a Triangle drawer class a Triangle input class and then a Triangle volume and a Triangle area.

File: Circle.cpp [hunk 1]
```diff
@@ -6,16 +6,14 @@
...
-#include "Volume.cpp"
-#include "Input.cpp"
-
+#include "ShapeManager.cpp"
 
 using namespace std;
-class Circle : public Shape
-{
+class Circle : public ShapeManager  
+//Responsibility: This class is responsible to provide implementation for draw(), inputDimensions, calculateArea(), and calculateVolume().
```
Comment (abbasshah17) [inline Circle.cpp]: You cannot use function names as the class's responsiblity. This is like defining a meaning of a word using the same word.

You need to focus on what drives the class's existence / creation or in other words what purpose the class is to fulfil.

The next step comes after: i.e. the purpose of the class should be single.

Once you've identified how the class violates that single responsibility principle, you need to state how it does so.

File: Enum.cpp [hunk 2]
```diff
@@ -19,7 +19,7 @@ enum Shapes
     INVALID
 };
 
-Shapes fromStringtoEnum(string shapeType)
+Shapes fromStringtoEnum(string shapeType)   // The function fromStringtoEnum() follows Single responsibility principle.
```
Comment (abbasshah17) [inline Enum.cpp]: We're currently discussing single responsibility principle for a class. The SOLID principle do not deal with low level implementations such as functions (even though it is a good practice to ensure that a function serves one purpose or job), this function's single responsibility is out of the scope of this task.

File: Input.cpp [hunk 3]
```diff
@@ -1,7 +1,8 @@
 #ifndef Input_cpp
 #define Input_cpp
 
-class Input 
+class Input  //Responsibility: This class provides a common interface for taking input.
```
Comment (abbasshah17) [inline Input.cpp]: Should be updated to

    //Responsibility: This class provides a common interface for taking input from user.

OR

    //Responsibility: This class provides a common interface for taking user-input.

File: Main.cpp [hunk 4]
```diff
@@ -14,21 +14,22 @@ int main()
     cout << "Enter a Shape" << endl;
     cin >> shapeType;
 
-    Shape *shapeDrawer = ShapeFactory ::create(shapeType);
-    shapeDrawer->inputDimensions();
-    double volume = shapeDrawer->calculateVolume();
-    double area = shapeDrawer->calculateArea();
-
-    if (shapeDrawer)
+    ShapeManager *shapeHandler = ShapeFactory ::create(shapeType);
```
Comment (abbasshah17) [inline Main.cpp]: Code refactoring was not part of this task. This PR should solely focus on identifying the responsibility of the classes.

File: Rectangle.cpp [hunk 5]
```diff
@@ -2,14 +2,15 @@
...
-#include "Area.cpp"
-#include "Volume.cpp"
-#include "Input.cpp"
+
+#include "ShapeManager.cpp"
 
 using namespace std;
 
-class Rectangle : public Shape
+class Rectangle : public ShapeManager
```
Comment (abbasshah17) [inline Rectangle.cpp]: See above [comment](https://github.com/Muhammadali2025222/C-Practice/pull/14/files#r1856597527).

File: ShapeManager.cpp [hunk 6]
```diff
@@ -0,0 +1,16 @@
+#ifndef ShapeManager_cpp
+#define ShapeManager_cpp
+
+#include "Area.cpp"
+#include "Volume.cpp"
+#include "Input.cpp"
+#include "Shape.cpp"
+
+class ShapeManager : public Area, public Volume, public Input, public Shape 
+// This class follows the Single Responsibility Principle.
```
Comment (abbasshah17) [inline ShapeManager.cpp]: The name / description of the responsibility of the class do not match.

The class name suggests, it *manages* `Shape`s. However, from the code, we can see that `ShapeManager` is a `Shape` itself.

File: Square.cpp [hunk 7]
```diff
@@ -2,14 +2,15 @@
...
-#include "Volume.cpp"
-#include "Input.cpp"
+
+#include "ShapeManager.cpp"
 
 using namespace std;
 
-class Square : public Shape
+class Square : public ShapeManager  
+//Responsibility: This class is responsible to provide implementation for draw(), inputDimensions, calculateArea(), and calculateVolume().
```
Comment (abbasshah17) [inline Square.cpp]: Same as before, check [comment](https://github.com/Muhammadali2025222/C-Practice/pull/14/files#r1856597527)

File: Triangle.cpp [hunk 8]
```diff
@@ -3,13 +3,13 @@
...
-#include "Shape.cpp"
-#include "Area.cpp"
-#include "Input.cpp"
+#include "ShapeManager.cpp"
 
 using namespace std;
 
-class Triangle : public Shape
+class Triangle : public ShapeManager  
+//Responsibility: This class is responsible to provide implementation for draw(), inputDimensions, calculateArea(), and calculateVolume().
```
Comment (abbasshah17) [inline Triangle.cpp]: Same as before, see [comment](https://github.com/Muhammadali2025222/C-Practice/pull/14/files#r1856597527).

**Review by abbasshah17 — CHANGES_REQUESTED** [review]
See [comment](https://github.com/Muhammadali2025222/C-Practice/pull/14/files#r1856597527)

You are still failing to understand what a class's responsibilities are and how to define them. 

Try this approach, without any code changes (such as `ShapeManager`):

1. Go through each class one by one, analyze how it interacts with other classes.
2. Analyze the current exposed contracts the class provides.
3. Write the responsibility of each class.
4. Again, one by one go to each class and determine whether the responsibility is singular in nature.
5. And then give your verdict of whether a class violates or follows the Single Responsibility principle.

In order to simplify this process. For now, focus on merely identifying the responsibilities of each class instead of focusing on whether it satisfies the Single Responsibility principle. That is, only go through points 1 - 3 mentioned above.

**Review by abbasshah17 — APPROVED** [review]

**Pull request merged**